### ISSUE TITLE
Support all (or at least more) forms of the functional API for Enum creation

### ISSUE THREAD

**Issue by tmke8** [microsoft/pyright#1127]
**Is your feature request related to a problem? Please describe.**

I'd like to use this form of Enum creation:
```python
B = Enum('B', ['p', 'q', 'r'])
```
but it is currently not supported by pyright.

A full list of allowed forms can be found here: python/mypy#2306

**Describe the solution you'd like**
I'd like pyright to accept the above code and to know about the members `B.p`, `B.q` and `B.r`. This is the behavior that `mypy` has.

**Additional context**
Pyright already supports this form:

```python
B = Enum('B', 'p q r'])
```
but I think this is visually not as easy to parse.

**Comment by erictraut** [comment]
There's already a much cleaner way to define enums if you have static enum labels, so I recommend you use that instead of these alternate forms. These alternate forms are generally only useful if your enum labels are dynamic. And in those cases, a static type checker won't be able to understand them anyway.

We could theoretically handle a specific case where you're using a list expression with string literals (as opposed to a variable that contains a list or a list constructor call), but it would involve a bunch of special-case code. I don't think it's worth it.

I'm going to close this as "won't fix", but other users can feel free to upvote or comment, and we might reconsider this decision in the future.

**Comment by matangover** [comment]
> These alternate forms are generally only useful if your enum labels are dynamic.

I disagree. It's useful to specify an enum as a one-liner without having to worry about the values.

**Comment by GBeauregard** [comment]
data point: mypy considered this a high priority issue https://github.com/python/mypy/issues/2306

If you're in strict mode pyright's errors are a bit harder to work around, but it might have only been high priority for mypy cause they error'd in default mode

**Comment by Kache** [comment]
Hoping pyright will also support the tuple/mapping syntax too!

```py
MyEnum = Enum('MyEnum', {'HA': 'ha', 'FOO': 'bar'})
```

**Comment by erictraut** [comment]
I've added support for the following Enum functional forms:
* Whitespace-separated names in a string: `Enum("name", "a b c")`
* Comma-separated names in a string: `Enum("name", "a,b,c")`
* List of names: `Enum("name", ["a", "b", "c"])`
* Tuple of names: `Enum("name", ("a", "b", "c"))`
* List of tuples: `Enum("name", [("a", 1), ("b", 2), ("c", 3)])`
* Tuple of tuples: `Enum("name", (("a", 1), ("b", 2), ("c", 3)))`
* Map: `Enum("name", {"a": 1, "b": 2, "c": 3})`

This will be included in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.304, which is now published. It will also be included in a future release of pylance.
